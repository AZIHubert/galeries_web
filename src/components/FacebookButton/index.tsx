import * as React from 'react';
import FacebookLogin, {
  ReactFacebookFailureResponse,
  ReactFacebookLoginInfo,
} from 'react-facebook-login';
import {
  useDispatch,
  useSelector,
} from 'react-redux';

import SocialMediaButton from '#components/SocialMediaButton';

import { fetchLoginFacebook } from '#store/actions';
import { loadingSelector } from '#store/selectors';

type Action = 'login' | 'signin';

interface FacebookButtonI {
  action?: Action;
}

const FacebookButton = ({
  action = 'login',
}: FacebookButtonI) => {
  const dispatch = useDispatch();
  const loading = useSelector(loadingSelector);

  const responseFacebook = async (
    faceBookResponse: ReactFacebookLoginInfo | ReactFacebookFailureResponse,
  ) => {
    dispatch(fetchLoginFacebook(faceBookResponse));
  };

  return (
    <FacebookLogin
      appId={process.env.FACEBOOK_ID!}
      callback={responseFacebook}
      fields="email, gender, name, picture.type(large)"
      render={(renderProps) => (
        <SocialMediaButton
          action={action}
          disabled={loading}
          marginBottom={10}
          marginBottomL={13}
          onClick={renderProps.onClick}
        />
      )}
    />
  );
};

export default FacebookButton;
