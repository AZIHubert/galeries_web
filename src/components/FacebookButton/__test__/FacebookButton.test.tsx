import * as React from 'react';
import renderer from 'react-test-renderer';
import {
  fireEvent,
  render,
  screen,
} from '@testing-library/react';

import FacebookButton from '../index';

const Container = () => {
  const [loading, setLoading] = React.useState(false);
  return (
    <FacebookButton
      loading={loading}
      setLoading={setLoading}
      type='login'
    />
  );
};

describe('Modal', () => {
  const setLoading = jest.fn;
  it('renders without crashing', () => {
    const tree = renderer.create(
      <FacebookButton
        loading={false}
        setLoading={setLoading}
        type='login'
      />,
    ).toJSON();
    expect(tree).toMatchSnapshot();
  });
  it('should not be disable on mount', () => {
    const { getByTestId } = render(
      <FacebookButton
        loading={false}
        setLoading={setLoading}
        type='signin'
      />,
    );
    const button = getByTestId('button');
    expect(button).not.toBeDisabled();
  });
  it('should display \'sign in with facebook if type is \'signin\'', () => {
    const { getByTestId } = render(
      <FacebookButton
        loading={false}
        setLoading={setLoading}
        type='signin'
      />,
    );
    const button = getByTestId('button');
    expect(button).toHaveTextContent('sign in with facebook');
  });
  it('should display \'log in with facebook if type is \'login\'', () => {
    const { getByTestId } = render(
      <FacebookButton
        loading={false}
        setLoading={setLoading}
        type='login'
      />,
    );
    const button = getByTestId('button');
    expect(button).toHaveTextContent('log in with facebook');
  });
  it('should be disabled after clicking', () => {
    const { getByTestId } = render(<Container />);
    const button = getByTestId('button');
    fireEvent.click(button);
    const buttonAfterClick = screen.getByTestId('button');
    expect(buttonAfterClick).toBeDisabled();
  });
});
