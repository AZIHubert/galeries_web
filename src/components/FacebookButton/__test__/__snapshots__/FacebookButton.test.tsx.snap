// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FacebookButton renders without crashing 1`] = `
<button
  data-testid="button"
  disabled={false}
  onClick={[Function]}
>
  <img
    alt="logo facebook"
    src=""
  />
  log in 
  with Facebook
</button>
`;
